DependencyInjection/Configuration.php

```diff
@@ -120,14 +120,18 @@ protected function configurePlugins(ArrayNodeDefinition $root)
             ->arrayNode('plugins')
                 ->addDefaultsIfNotSet()
                 ->children()
-
                     ->arrayNode('authentication')
-                    ->canBeEnabled()
-                        ->children()
-                            ->scalarNode('authentication')
-                                ->info('This must be a service id to a service implementing Http\Message\Authentication')
-                                ->isRequired()
-                                ->cannotBeEmpty()
+                        ->useAttributeAsKey('name')
+                        ->prototype('array')
+                            ->children()
+                                ->enumNode('type')
+                                    ->values(['basic', 'bearer', 'wsse'])
+                                    ->isRequired()
+                                    ->cannotBeEmpty()
+                                ->end()
+                                ->scalarNode('username')->end()
+                                ->scalarNode('password')->end()
+                                ->scalarNode('token')->end()
```

Comment:
If we add type `service` then you could use the authentication plugin just as before. It just require you to do a minor config change. 

Before: 
```yaml
// config.yml
httplug:
    plugins:
        authentication:
             authentication: 'my_authentication_service'
    clients:
        my_client:
            factory: 'httplug.factory.guzzle6'
            plugins: ['httplug.plugin.authentication']
```

After:
```yaml
// config.yml
httplug:
    plugins:
        authentication:
            acme:
                type: 'service'
                service: 'my_authentication_service'
    clients:
        my_client:
            factory: 'httplug.factory.guzzle6'
            plugins: ['httplug.plugin.authentication.acme']
```